src/cracking.py

```diff
@@ -0,0 +1,307 @@
+
+			with console.status(f"[bold magenta] Searching [reversed] {hash_ctext}") as status:
+				results = self.threaded_search(future, supported_searchers)
+				for k,v in results.items():
+					console.print(f"[bold blue] {k} : [bold red] {v}")
+
+	def threaded_search(self, future, supported_searchers):
+
+		processes = []
+		results = {}
+
+		with ThreadPoolExecutor(max_workers=4) as executor:
+			for search in supported_searchers:
+				processes.append(executor.submit(self.call_searcher, search, future))
+
+		for d in processes:
+			results.update(d.result())
+
+		return results
+	
+	def call_searcher(self, search, future):
+		try:
+			return {type(search).__name__:search.crack(future)}
+		except Exception as e:
+			logger.warn(f"{type(search).__name__} [{e}]")
+			return {type(search).__name__:False}
+
+
+'''
+class hashtoolkit:
+
+	supports = set(["md5", "ntlm"])
+	moduels = ["cloudscraper", "requests"]
+	offline = False
+
+	def crack(self, hash):
+		try:
+			Scraper = cloudscraper.create_scraper()
+			HTML = Scraper.get(
+				f"https://hashtoolkit.com/decrypt-hash/?hash={hash[0]}", timeout=1
+			).text.splitlines()
+		except:
+			logger.warn("Couldn't connect to hashtoolkit")
+			return False  # >> Remember to log this
+
+		for i in range(len(HTML)):
+			if "/generate-hash/?text=" in HTML[i]:
+
+				output = HTML[i].partition("?text=")[2].split('">')[0]
+
+				if "</a></span>" in output or 'title"' in output:
+					logger.warn("Hashtoolkit couldnt crack hash")
+					return False  # >> And this
+				else:
+					return output
+
+		return False  # >> Dont think this would ever happen but log this
+-- removing? '''
+
+# Bug Fixing - API / Domain is currently down
+class hashsorg:
+
+	supports = set(["md5", "NTLM", "SHA-1"])
+	moduels = ["requests"]
+	offline = False
+
+	# How do we implment the API key?
+
+	def crack(self, hash):
+		try:
+			request = requests.get(
+				f"https://hashes.org/api.php?key={key}&query={hash[0]}", timeout=1
+			).text
+		except:
+			logger.warn("Couldn't connect to hashesorg")
+			return False
+
+		# Check for false positive
+
+		if "null" in request:
+			logger.warn("Couldnt connect to hashesorg")
+			return False
+		if "" == request:
+			logger.warn("Hash seems to be a plain???")
+			return False
+
+		output = request.split('":"')[2].split('","')[0]
+
+		# Check for Hex
+
+		if "$HEX[" in output:
+			return bytearray.fromhex(
+				final_output[5 : len(final_output) - 1]
+			).decode()  # Partions it so that it only contains hex and then decodes it into ASCII
+
+		return output
+
+# Need to support IDS but will wait for format from name-that-hash
+class hashcat:
+
+	supports = set(["md5", "NTLM", ""])
+	moduels = ["subprocess"]
+	offline = True
+
+	# How do we implement the types?? (IDs) and wordlist??
+
+	def crack(self, hash):
+		for possible_type in range(len(ids)):
+			command = f"hashcat64.exe -a 0 -m {ids[possible_type]} {hash[0]} {wordlist}"  # >> Can change for better optimization etc.....

```

Comment:
` # >> Can change for better optimization etc.....`

This comment is kinda pointless? :)